### ISSUE TITLE
SC2311 is incorrectly not triggered with `set -o pipefail`

### ISSUE THREAD

**Issue by rolfschr** [koalaman/shellcheck#3367]
#### For bugs with existing features

- Rule Id SC2311
- My shellcheck version 0.11.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2311)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

The following code will echo "This is a bug" inside will_fail when invoked in a subshell because BASH implicitly disables `set -e` their. This is correctly reported in SC2311. However, when `set -o pipefail` is set, than SC2311 is _not_ triggered _although_ th problem persists. Here is the code and some screenshots: 

```bash
> cat test.sh
#!/usr/bin/env bash

#shellcheck enable=check-set-e-suppressed
set -ex

# When setting `set -o pipefail`, SC2311 is _not_ triggered!
# set -o pipefail

will_fail() {
  false
  echo "This is a bug! This should never get executed!"
}

# This will invoke the `echo`. Exactly like SC2311 predicted!
version=$(will_fail)
echo "$version" >/dev/null

# This works as expected: The `echo` will never get executed.
will_fail
```

Without (!) `set -o pipefail`, shellsheck reports the problem:
<img width="1585" height="125" alt="Image" src="https://github.com/user-attachments/assets/35cdea4d-e490-4a5c-b38d-9c818562fb09" />

With `set -o pipefail`, shellcheck doesn't complain _but_ the problem (=set -e is implicitly disabled) persists:

<img width="587" height="176" alt="Image" src="https://github.com/user-attachments/assets/8dd0248d-5589-4b9a-9ff0-778bbf75be68" />

I'm no BASH expert, I found this out the hard way. I am not sure whether Shellcheck is "wrong" here but it is unexpected output at least for me.
